k8s/upgrade/src/upgrade_resources/upgrade.rs

```
@@ -805,3 +805,20 @@ impl ImageProperties {
         self.pull_policy.clone()
     }
 }
+
+/// Return the installed version.
+pub async fn get_source_version(ns: &str) -> Result<String, Error> {
+    match get_deployment_for_rest(ns).await {
+        Ok(deployment) => match &deployment.metadata.labels {
+            Some(label) => match label.get(HELM_RELEASE_VERSION_LABEL) {
+                Some(value) => Ok("v".to_owned() + value),
+                None => Err(Error::UpgradeEventNotPresent),
+            },
+            None => Err(Error::UpgradeEventNotPresent),
+        },
+        Err(e) => {
+            eprintln!("Failed in fetching deployment {e:?}");
+            std::process::exit(1);
```

This should return an error like the others?